src/ngraph_builder.cc

```diff
@@ -3523,12 +3523,15 @@ static Status TranslateStridedSliceOp(
                               // type inference of return type
     } else if (idx > dim - 1) {
       return (int)dim;
-    }  // The next case handles the clamping (differently for inclusive and
-       // exclusive cases)
-    else if (idx + dim <
-             0) {  // careful not to do idx < -dim (since dim is unsigned)
+    } else if (static_cast<int>(idx + dim) < 0) {
```

Comment:
there is a `idx + dim >= 0` in line 3520 as well, which adds an int and a size_t. Do you want to cast that as well.
I think clockwork complains for line 3520 as well: https://klocwork-ir2.devtools.intel.com:8080/review/insight-review.html#issuedetails_goto:offset=0,problemid=9450,project=ngraph_tf,searchquery=state%253A%252BNew,sortcolumn=id,sortdirection=ASC,start=0,view_id=8

Also, I wonder should we cast dim to int (like line 3522) or cast the sum to int? casting the sum to int works, but in that case we are still adding 2 different types. casting dim to int first would add same types.

Reply:
Yes - I think so.